src/python/pants/engine/subsystem/native.py

```diff
@@ -166,7 +168,7 @@
     ExecutionStat execution_execute(RawScheduler*);
     RawNodes* execution_roots(RawScheduler*);
 
-    void validator_run(RawScheduler*, TypeId*, uint64_t);
+    Value* validator_run(RawScheduler*, TypeId*, uint64_t);
```

Comment:
Generally, for things that don't contain a rust-side allocation, we'd just return the whole struct (ie, copy it to python). Here, I'm pretty sure this allocation will be leaked (because the `Box` it is in can't be recovered afterward).

So probably just return a `Value` by-value (and have the python side drop its handle?).